services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/cache/LocalCacheImpl.java

```diff
@@ -72,14 +73,36 @@ public void addSubnetPorts(NetworkConfiguration networkConfig) throws Exception
 
                 InternalSubnetPorts subnetPorts = subnetPortsMap.get(subnetId);
                 if (subnetPorts == null) {
-                    SubnetEntity subnetEntity = getSubnetEntity(networkConfig, fixedIp.getSubnetId());
+                    InternalSubnetEntity subnetEntity = getSubnetEntity(networkConfig, fixedIp.getSubnetId());
+
                     subnetPorts = new InternalSubnetPorts();
                     subnetPorts.setSubnetId(subnetId);
                     subnetPorts.setGatewayPortMac(subnetEntity.getGatewayPortDetail().getGatewayMacAddress());
                     subnetPorts.setGatewayPortIp(subnetEntity.getGatewayIp());
                     subnetPorts.setGatewayPortId(subnetEntity.getGatewayPortDetail().getGatewayPortId());
+                    subnetPorts.setName(subnetEntity.getName());
+                    subnetPorts.setCidr(subnetEntity.getCidr());
+                    subnetPorts.setVpcId(subnetEntity.getVpcId());
+                    subnetPorts.setTunnelId(subnetEntity.getTunnelId());
+                    subnetPorts.setDhcpEnable(subnetEntity.getDhcpEnable());
                     subnetPorts.setPorts(new ArrayList<>());
 
+                    List<InternalRouterInfo> routers = networkConfig.getInternalRouterInfos();
+                    if (routers != null && routers.size() > 0) {
+                        String router_id = "";
+                        for (InternalRouterInfo router : routers) {
+                            List<InternalSubnetRoutingTable> subnetRoutingTables = router.getRouterConfiguration().getSubnetRoutingTables();
```

Comment:
Could subnetRoutingTables be NULL?

Reply:
add one line of code to check if `subnetRoutingTables` is null.